components/site-search/site-search.js

```diff
@@ -84,10 +84,15 @@ Search.prototype.resultTemplate = function (result) {
     const resultTitle = result.title
     element.textContent = resultTitle
 
-    if (result.dateString || result.section) {
+    if (result.hasFrontmatterDate || result.section) {
       const section = document.createElement('span')
       section.className = 'app-site-search--section'
-      section.innerHTML = result.dateString || result.section
+
+      if (result.hasFrontmatterDate && result.section) {
+        section.innerHTML = `${result.section}<br>${result.date}`
```

Comment:
This will dictate the layout of how the results are displayed in the frontend to the user, this is probably fine but worth noting in the changelog / release notes.

I did wonder whether there's scope for a future discussion around this for how results are displayed. Currently this would give a user a result like:
```
Post title
Section name
19 May 2023
```

But it's possible a user might prefer for example:
```
Post title
Section name - 19 May 2023
```

Not a blocker for me by any means, just maybe worth considering for future enhancements.

Reply:
One reason for having section and date on separate lines is that the search results list has a fixed width; ‘Apply for teacher training - 22 February 2022’ would end up partially cut off, for example.

It’s always difficult to find the right balance between providing good defaults, and having everything be configurable (which can introduce bugs and make the project harder to maintain).

Hopefully this change fixes the immediate bug, and ensures this results display as originally intended. We can then look to see if further options are needed later.